### PR TITLE
[IMP] hr_public_holidays: Display those in calendar views

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Very good addition!

File: hr_public_holidays/__manifest__.py [hunk 1]
```diff
@@ -18,6 +18,9 @@
         'security/ir.model.access.csv',
         'views/hr_public_holidays_view.xml',
         'wizards/public_holidays_next_year_wizard.xml',
+
```
Comment (tafaRU) [inline hr_public_holidays/__manifest__.py]: why this blank line?

**Review by tafaRU — CHANGES_REQUESTED** [review]
Thanks a lot for your PR! I tested on runbot :+1: 
Could you also please update the module description in the README file accordingly and increase the module version?

File: hr_public_holidays/static/src/js/holidays_highlighter.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+calendarView.include({
+    get_fc_init_options: function () {
+        var res = this._super();
+
+        var oldRenderHandler = res['viewRender'];
+        // extend view render callback w/ highlighting functionality
+        res['viewRender'] = function (view) {
+            oldRenderHandler.apply(this, arguments);  // read as super()
+            var visibleDays = $('.o_calendar_view .fc-day');
+            var publicHolidays = hrHolidays.call(
```
Comment (simahawk) [inline hr_public_holidays/static/src/js/holidays_highlighter.js]: these records should be loaded at `willStart` and stored on `this`

File: hr_public_holidays/static/src/js/holidays_highlighter.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                'search_read', [[], ['date']]);
+            var holidayColor = irConfigParameter.call(
+                'get_param', ['calendar.public_holidays_color']);
+            $.when(publicHolidays, visibleDays, holidayColor)
+                .then(function (publicHolidays, visibleDays, holidayColor) {
+                    // as a result of `search_read` call is the JS object
+                    // (dictionary), it's still our duty to clean up results
+                    var publicHolidaysArr = publicHolidays.map(x => x['date']);
```
Comment (simahawk) [inline hr_public_holidays/static/src/js/holidays_highlighter.js]: this should be done in the willStart call

File: hr_public_holidays/static/src/js/holidays_highlighter.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+calendarView.include({
+    get_fc_init_options: function () {
+        var res = this._super();
+
+        var oldRenderHandler = res['viewRender'];
+        // extend view render callback w/ highlighting functionality
+        res['viewRender'] = function (view) {
+            oldRenderHandler.apply(this, arguments);  // read as super()
+            var visibleDays = $('.o_calendar_view .fc-day');
```
Comment (simahawk) [inline hr_public_holidays/static/src/js/holidays_highlighter.js]: this.$

File: hr_public_holidays/static/src/js/holidays_highlighter.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
+odoo.define('hr_public_holidays.holidays_highlighter', function (require) {
+'use strict';
+
+var calendarView = require('web_calendar.CalendarView');
+
+var Model = require('web.Model');
+var hrHolidays = new Model('hr.holidays.public.line');
```
Comment (simahawk) [inline hr_public_holidays/static/src/js/holidays_highlighter.js]: you should instantiate them where you use them

**Review by simahawk — CHANGES_REQUESTED** [review]

File: hr_public_holidays/static/src/js/holidays_highlighter.js (hunk 4 above)
Comment (ghost) [reply hr_public_holidays/static/src/js/holidays_highlighter.js]: sadly enough, `TypeError: this.$ is not a function` :wink:
upd: though, `var self = this` approach solves it

**Comment by ghost** [comment]
@tafaRU @simahawk comments addressed :wink:

File: hr_public_holidays/__manifest__.py [hunk 6]
```diff
@@ -16,6 +16,8 @@
     ],
     'data': [
         'security/ir.model.access.csv',
+        'data/ir_config_parameter.xml',
+        'static/src/xml/assets.xml',
```
Comment (simahawk) [inline hr_public_holidays/__manifest__.py]: sorry, missed this path: should not be in static. `templates/assets.xml` works better :)

**Review by simahawk — APPROVED** [review]
small path change. Other than this, LGTM!

File: hr_public_holidays/__manifest__.py (hunk 6 above)
Comment (ghost) [reply hr_public_holidays/__manifest__.py]: yup, seems neat

File: hr_public_holidays/data/ir_config_parameter.xml [hunk 7]
```diff
@@ -0,0 +1,8 @@
+<odoo>
+
+  <record id="public_holiday_color" model="ir.config_parameter">
+    <field name="key">calendar.public_holidays_color</field>
+    <field name="value">#cc9bef</field>
```
Comment (tarteo) [inline hr_public_holidays/data/ir_config_parameter.xml]: Can you change the default color to a 'Odoo' color? This way it feels by default more natural.

**Review by tarteo — CHANGES_REQUESTED** [review]
Cool addition! Is it possible to add the name of the public holiday in the highlighted area?

I see a problem if a holiday is country specific. And an employee of another country sees the public holiday in their calendar.
In my opinion creating a hr.holidays for every employee is a better solution. This way you can easily impose some logic etc to it.

File: hr_public_holidays/data/ir_config_parameter.xml (hunk 7 above)
Comment (ghost) [reply hr_public_holidays/data/ir_config_parameter.xml]: what exactly do you call an 'Odoo' color? :slightly_smiling_face:

Comment (tarteo) [reply hr_public_holidays/data/ir_config_parameter.xml]: Like the purple of Odoo #7c7bad

**Review by tafaRU — APPROVED** [review]

File: hr_public_holidays/data/ir_config_parameter.xml (hunk 7 above)
Comment (simahawk) [reply hr_public_holidays/data/ir_config_parameter.xml]: the purple one maybe?

Comment (ghost) [reply hr_public_holidays/data/ir_config_parameter.xml]: isn't the original one purple enough already? :smile: 
image

File: hr_public_holidays/data/ir_config_parameter.xml [hunk 8]
```diff
@@ -0,0 +1,8 @@
+<odoo>
```
Comment (grindtildeath) [inline hr_public_holidays/data/ir_config_parameter.xml]: Please add XML declaration, and probably `noupdate="1"`.

**Review by grindtildeath — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Any chances of this being forward-ported to v11?

**Comment by pedrobaeza** [comment]
Can this please be forward-ported to v11?

**Pull request merged**